src/write.rs

```diff
@@ -143,7 +143,7 @@ impl<W: Write> SaveWriter<W> {
 
         // write bricks and components
         {
-            let mut vec = vec![];
+            let mut vec = Vec::with_capacity(self.data.bricks.len() * 8);
```

Comment:
I like your reasoning for 8 bytes -- could you also throw this into a `const` in this module and add a doc comment explaining your reasoning? You can link to this PR if you want. I think it would be helpful for future readers